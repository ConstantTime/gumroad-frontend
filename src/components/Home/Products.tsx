import Typography from "@material-ui/core/Typography";
import axios from "axios";
import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { IProduct } from "../../api/responses/types";
import { ApiEndpoints, SERVER_URL } from "../../constants/Constants";
import "./products.css";

const Products: React.FC = () => {
  const [products, setProducts] = useState<IProduct[]>([]);

  useEffect(() => {
    axios
      .get<IProduct[]>(SERVER_URL + ApiEndpoints.FETCH_PRODUCTS, {
        method: "GET",
      })
      .then((response) => {
        setProducts(response.data);
      })
      .catch((response) => {
        console.log("Error:", response);
      });
  }, []);

  return (
    <div>
      <Typography variant="h3" component="h2" align="center">
        <div className="products-list-heading">Gumroad Products</div>
      </Typography>
      <div className="products-list">
        {products.map((product, index) => (
          <Typography
            key={`${index}`}
            variant="h5"
            component="h2"
            align="center"
          >
            <Link to={`/product/${product.productId}`}>{product.name}</Link>
          </Typography>
        ))}
      </div>
    </div>
  );
};

export default Products;
