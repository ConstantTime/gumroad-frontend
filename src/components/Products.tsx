import Typography from "@material-ui/core/Typography";
import axios from "axios";
import React, { useEffect, useState } from "react";
import { Product } from "../api/responses/types";
import { ApiEndpoints, SERVER_URL } from "../constants/Constants";

const Products: React.FC = () => {
  const [products, setProducts] = useState<Product[]>([]);

  useEffect(() => {
    axios
      .get<Product[]>(SERVER_URL + ApiEndpoints.FETCH_PRODUCTS, {
        method: "GET",
      })
      .then((response) => {
        setProducts(response.data);
      })
      .catch((response) => {
        console.log(response);
      });
  }, []);

  return (
    <Typography variant="h3" component="h2" align="center">
      {products.map((product) => {
        return <div>{product.name}</div>;
      })}
    </Typography>
  );
};

export default Products;
