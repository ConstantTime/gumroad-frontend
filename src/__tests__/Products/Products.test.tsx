import { shallow } from "enzyme";
import React from "react";
import Products from "../../components/Home/Products";

describe("Products", () => {
  it("should render the component", () => {
    const wrapper = shallow(<Products />);
    expect(wrapper).toMatchSnapshot();
  });
});
