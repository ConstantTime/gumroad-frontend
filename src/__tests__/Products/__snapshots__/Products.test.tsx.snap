// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products should render the component 1`] = `
<div>
  <WithStyles(ForwardRef(Typography))
    align="center"
    component="h2"
    variant="h3"
  >
    <div
      className="products-list-heading"
    >
      Gumroad Products
    </div>
  </WithStyles(ForwardRef(Typography))>
  <div
    className="products-list"
  />
</div>
`;
